From 7b30ea927d8a90b01996d6d840977eb27386a6b4 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Wed, 24 Aug 2016 20:27:59 +0300
Subject: [PATCH 0436/1345] pp3: Remove default shaping and drop for EMAC0

commit  b150043cb5eb46d0caca274789b3ccb7d2ebc34b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

EMAC0 TM port (4) have maximum throughput 10G
and configured to no drop profile

Change-Id: Ica10faa4cb730eff30a089bc0428a5e2b885dcd6
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32182
Reviewed-by: Max Kovalenko <maxk@marvell.com>
Reviewed-by: Yan Markman <ymarkman@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/pp3/tm/mv_tm.c |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/marvell/pp3/tm/mv_tm.c b/drivers/net/ethernet/marvell/pp3/tm/mv_tm.c
index db686f2..dc3b603 100644
--- a/drivers/net/ethernet/marvell/pp3/tm/mv_tm.c
+++ b/drivers/net/ethernet/marvell/pp3/tm/mv_tm.c
@@ -564,8 +564,6 @@ int tm_cfg1(void)
 	struct tm_port_params p_params;
 	enum tm_port_bw port_speed = TM_10G_PORT;
 
-	uint32_t cbw = MV_TM_2HG_BW; /* shaping rate 2.5G */
-	uint16_t dp_index = 1; /* drop profile index */
 	uint8_t port_index;
 	uint8_t i;
 	int rc;
@@ -683,8 +681,7 @@ int tm_cfg1(void)
 		}
 
 		if (port_index == TM_A0_PORT_EMAC0) {
-			/* 2.5G "shaping" drop */
-			p_params.wred_profile_ref = (uint8_t) dp_index;
+			p_params.wred_profile_ref = TM_NO_DROP_PROFILE;
 			rc = tm_create_port(hndl, port_index, port_speed, &p_params,
 				num_of_c_nodes,
 				(num_of_b_nodes*num_of_c_nodes),
@@ -724,9 +721,6 @@ int tm_cfg1(void)
 			goto err_out;
 	}
 
-	/* 2.5G shaping on port EMAC0 */
-	rc = tm_set_min_shaping(hndl, P_LEVEL, TM_A0_PORT_EMAC0, cbw);
-
 err_out:
 	if (rc == 0)
 		pr_info("TM cfg1 Scenario Completed Successfuly\n");
-- 
1.7.9.5

